common/util.cc

```diff
@@ -257,6 +257,7 @@ bool starts_with(const std::string &s1, const std::string &s2) {
 }
 
 bool ends_with(const std::string &s1, const std::string &s2) {
+  if (s2.size() > s1.size()) return false;
```

Comment:
it's not obvious that's what this check is for. what about using `strncmp`?

Reply:
looks like we always need to check the size before strcmp or strncmp